app/bundles/CoreBundle/Shortener/Shortener.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+
+namespace Mautic\CoreBundle\Shortener;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Shortener\CustomApi\CustomApiShortener;
+
+class Shortener
+{
+    public const SHORTENER_SERVICE = 'shortener_service';
+
+    private CoreParametersHelper $coreParametersHelper;
+
+    private CustomApiShortener $customApiShortener;
+
+    /**
+     * @var ShortenerServiceInterface[]
+     */
+    private array $services = [];
+
+    public function __construct(CoreParametersHelper $coreParametersHelper, CustomApiShortener $customApiShortener)
+    {
+        $this->coreParametersHelper = $coreParametersHelper;
+        $this->customApiShortener   = $customApiShortener;
+    }
+
+    public function addService(string $id, ShortenerServiceInterface $shortener)
+    {
+        $this->services[$id] = $shortener;
+    }
+
+    public function getService(string $name = null): ShortenerServiceInterface
+    {
+        if (!$name) {
+            $name = $this->coreParametersHelper->get(self::SHORTENER_SERVICE);
+        }
+
+        if (isset($this->services[$name])) {
+            return $this->services[$name];
+        }
+
+        throw new \InvalidArgumentException(sprintf('There is not a shortener service  %s', $name));
+    }
+
+    /**
+     * @return ShortenerServiceInterface[]
+     */
+    public function getServices(): array
+    {
+        return $this->services;
+    }
+
+    public function getEnabledServices(): array
+    {
+        $enabledServices = [];
+
+        foreach ($this->services as $name => $service) {
+            if ($service->isEnabled()) {
+                $enabledServices[$name] = $service;
+            }
+        }
+
+        return $enabledServices;
```

Comment:
Just geeking out here but this seems like a perfect example for https://www.php.net/manual/en/function.array-filter.php
```suggestion
        return array_filter($this->services, fn ($service) => $service->isEnabled());
```